### PR TITLE
jsreport server deployment in azure

### PR DESCRIPTION
**Pull request by irajbalakrish**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: jsreport-linux-appservice-docker/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Provision jsreport service in Azure
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fazure%2Fazure-quickstart-templates%2Fmaster%2Fjsreport-linux-appservice-docker%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fjsreport-linux-appservice-docker%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template allows you to deploy jsreport, a neat and elegant library / service for drawing PDF Reports using HTML and handlebars Templates. You can pick from a variety of recipes aka Algorithms to generate PDF’s using engines like PhantomPDF, Electron,Chromium,WkhtmltoPDF etc.., This Template will create a Linux App Service + StorageAccount, Pull the Jsreports 2.1.0-full version build from Docker Registry and Configure Authentication and Embed All the settings required to connect to a Storage Account in order to Persist the Templates.
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/README.md]: fix punctuation, spacing, typos

File: jsreport-linux-appservice-docker/README.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+The default credentials to log into jsreport service will be 'username:admin,password:password'
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/README.md]: the creds need to be passed in by the user

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      ],
+      "metadata": {
+        "description": "Describes the Storage Account Type"
+      }
+    }
+  },
+  "variables": {
+    "servicePlanName": "[concat('serviceplan',uniqueString(resourceGroup().id))]",
+    "StorageName": "[concat('jsreports', uniqueString(resourceGroup().id))]",
+    "storageAccountApiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: remove this and use a literal apiVersion

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    "storageId": "[resourceId('Microsoft.Storage/storageAccounts', variables('StorageName'))]",
+    "storageConnectionStringPrefix":
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: use reference to get the uri - reference is not allowed in the vars section so you'll need to move this inline...

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      "[concat('DefaultEndpointsProtocol=https;AccountName=',variables('StorageName'),';EndpointSuffix=core.windows.net',';AccountKey=')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Web/serverfarms",
+      "name": "[variables('servicePlanName')]",
+      "apiVersion": "2016-03-01",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: update apiVersion

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+      "properties": {
+        "workerSizeId": 0,
+        "reserved": true,
+        "hostingEnvironment": ""
+      }
+    },
+    {
+      "type": "Microsoft.Web/sites",
+      "name": "[parameters('appName')]",
+      "apiVersion": "2016-03-01",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: update apiVersion

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        "serverFarmId": "[resourceId('Microsoft.Web/serverfarms', variables('servicePlanName'))]"
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Web/serverfarms', variables('servicePlanName'))]"
+      ],
+      "resources": [
+        {
+          "name": "appsettings",
+          "type": "config",
+          "apiVersion": "2016-03-01",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: update apiVersion

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        "workerSizeId": 0,
+        "reserved": true,
+        "hostingEnvironment": ""
+      }
+    },
+    {
+      "type": "Microsoft.Web/sites",
+      "name": "[parameters('appName')]",
+      "apiVersion": "2016-03-01",
+      "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+            "extensions_fsStoreAzureStoragePersistence_accountKey": "[listkeys(variables('storageId'), variables('storageAccountApiVersion')).key1]",
+            "extensions_fsStoreAzureSbSync_connectionString": "[concat(variables('storageConnectionStringPrefix'), listkeys(variables('storageId'), variables('storageAccountApiVersion')).key1)]"
+          }
+        }
+      ]
+    },
+    {
+      "name": "[variables('StorageName')]",
+      "type": "Microsoft.Storage/storageAccounts",
+      "location": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: Add location parameter - see: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#parameters

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+      "apiVersion": "2016-01-01",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: update apiVersion

File: jsreport-linux-appservice-docker/metadata.json [hunk 12]
```diff
@@ -0,0 +1,8 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "itemDisplayName": "jsreport in azure",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/metadata.json]: capitalize "Azure"

File: jsreport-linux-appservice-docker/metadata.json [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+  "description": "This template allows you to deploy jsreport service on a Linux AppService.The current version of the image is 2.1.0 and pulled through Docker Hub Public Registry",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/metadata.json]: fix spacing and punctuation

Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/metadata.json]: Applied Changes

File: jsreport-linux-appservice-docker/metadata.json (hunk 12 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/metadata.json]: Applied Changes

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 11 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/azuredeploy.json]: updated to 2017-10-01

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 10 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/azuredeploy.json]: Applied Changes to "location": "[parameters('location')]",

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 9 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/azuredeploy.json]: Applied Changes to "location": "[parameters('location')]",

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 8 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/azuredeploy.json]: updated to  "apiVersion": "2016-08-01",

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 7 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/azuredeploy.json]: "apiVersion": "2016-08-01",

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 6 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/azuredeploy.json]: updated Location to parameter

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 5 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/azuredeploy.json]: Updated to  "apiVersion": "2016-09-01",

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 3 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/azuredeploy.json]: Removed this variable and used explicit api versioning

File: jsreport-linux-appservice-docker/README.md (hunk 1 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/README.md]: Applied Changes

File: jsreport-linux-appservice-docker/README.md (hunk 2 above)
Comment (irajbalakrish) [reply jsreport-linux-appservice-docker/README.md]: This is now removed from README

File: jsreport-linux-appservice-docker/azuredeploy.json [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+    "appName": {
+      "defaultValue": "[concat('jsreportsapp',uniqueString(resourceGroup().id))]",
+      "type": "string",
+      "metadata": {
+        "description": "Name of the WebApp Running jsreport"
+      }
+    },
+    "location": {
+      "type": "string",
+      "metadata": {
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.json]: add:

"defaultValue": "[resourceGroup().location]"

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 3 above)
Comment (bmoore-msft) [reply jsreport-linux-appservice-docker/azuredeploy.json]: remove the variable

File: jsreport-linux-appservice-docker/azuredeploy.json (hunk 4 above)
Comment (bmoore-msft) [reply jsreport-linux-appservice-docker/azuredeploy.json]: remove this

File: jsreport-linux-appservice-docker/azuredeploy.parameters.json [hunk 15]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "appName": {
+      "value": "GEN-UNIQUE"
+    },
+    "location": {
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/azuredeploy.parameters.json]: remove this when defaultValue is added to the location param

File: jsreport-linux-appservice-docker/metadata.json [hunk 16]
```diff
@@ -0,0 +1,8 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "itemDisplayName": "jsreport in Azure",
+  "description": "This template allows you to deploy jsreport on a Linux AppService.The build version is 2.1.0 and pulled through Docker Public Registry",
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/metadata.json]: add spacing after punctuation

File: jsreport-linux-appservice-docker/README.md [hunk 17]
```diff
@@ -0,0 +1,14 @@
+# Provision jsreport service in Azure
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fazure%2Fazure-quickstart-templates%2Fmaster%2Fjsreport-linux-appservice-docker%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fjsreport-linux-appservice-docker%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template allows you to deploy jsreport, an elegant service for drawing PDF Reports using HTML and handlebars Templates. You can pick from a variety of recipes aka algorithms to generate PDF using engines like PhantomPDF, Electron,Chromium,WkhtmltoPDF etc. This Template will create a Linux App Service, StorageAccount and shall pull jsreport 2.1.0-full version from Docker Registry, configure authentication and embed all the settings required to connect to a Storage Account in order to Persist the templates.
```
Comment (bmoore-msft) [inline jsreport-linux-appservice-docker/README.md]: add correct spacing after punctuation

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.

**Pull request closed without merging**